File: scala-server/app/com/bolour/plane/scala/domain/Point.scala
/*
 * Copyright 2017 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */
package com.bolour.plane.scala.domain

import com.bolour.plane.scala.domain.Axis.Axis

case class Point(row: Int, col: Int) {
  def colinearPoint(axis: Axis, direction: Int)(steps: Int): Point = {
    val offset = direction * steps
    axis match {
      case Axis.X => Point(row, col + offset)
      case Axis.Y => Point(row + offset, col)

    }
  }

  def adjPoint(axis: Axis, direction: Int): Point = colinearPoint(axis, direction)(1)
}

File: java-client/src/main/java/com/bolour/boardgame/client/domain/Point.java
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */

package com.bolour.boardgame.client.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Point {
    public final int row, col;

    @JsonCreator
    public Point(
      @JsonProperty("row") int row,
      @JsonProperty("col") int col
    ) {
        this.row = row;
        this.col = col;
    }
}
